src/sonic-config-engine/sonic_device_util.py

```diff
@@ -29,6 +29,39 @@ def get_machine_info():
             machine_vars[tokens[0]] = tokens[1].strip()
     return machine_vars
 
+def get_num_npus():
+   platform = get_platform_info(get_machine_info)
+   num_npus = 1
+   asic_conf_file_path = os.path.join('/usr/share/sonic/device', platform, 'asic.conf')
+   if not os.path.isFile(asic_conf_file_path):
+	return num_npus
+   with open(asic_conf_file_path) as asic_conf_file:
+	for line in asic_conf_file:
+	    tokens = line.split('=')
+            if len(tokens) < 2:
+               continue      
+	    if token[0].lower() == 'num_asic':
+		num_npus = token[1].strip()
+   		return num_npus
+
+def get_namespaces():
```

Comment:
i do not see this function used anywhere else in you code, in the multi npu system, why not use asic.conf to derive the namepsace you have on the system? why use ip netns list to query?

Reply:
The plan is use this  function by CLI commands. 
Today the namespace are names as "asicX" , it could change in the future so didn't want to generate the name from asic.conf here.